bin.js

```diff
@@ -8,27 +8,32 @@ if (argv.length === 0) {
   process.exit(1)
 }
 const wantedPM = argv[0]
+const wantedVersion = arg[1]
+
 if (wantedPM !== 'npm' && wantedPM !== 'pnpm' && wantedPM !== 'yarn') {
   console.log(`"${wantedPM}" is not a valid package manager. Available package managers are: npm, pnpm, or yarn.`)
   process.exit(1)
 }
 const usedPM = whichPMRuns()
-if (usedPM && usedPM.name !== wantedPM) {
+if (usedPM && usedPM.name !== wantedPM && usedPM.version !== wantedVersion) {
```

Comment:
what about when the version is not specified? In that case it shouldn't be checked, right?
```suggestion
if (usedPM && (usedPM.name !== wantedPM || wantedVersion && usedPM.version !== wantedVersion)) {
```